model-optimizer/mo/ops/broadcast_test.py

```diff
@@ -0,0 +1,59 @@
+"""
+ Copyright (C) 2018-2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import unittest
+
+import numpy as np
+from generator import generator, generate
+
+from mo.front.common.partial_infer.utils import int64_array
+from mo.graph.graph import Node
+from mo.ops.broadcast import Broadcast
+from mo.utils.unittest.graph import build_graph, valued_const_with_data, regular_op_with_empty_data, \
+    shaped_data
+
+
+@generator
+class BroadcastTest(unittest.TestCase):
+    @generate(*[
+        ([1], [3, 3], [[1, 1, 1], [1, 1, 1], [1, 1, 1]], None, 'numpy', True),
+        ([1], [3, 3], [3, 3], None, 'numpy', False),
+        ([1], [2, 2], [[1, 1]], [0], 'explicit', True),
+        ([1], [2, 2], [1, 2], [0], 'explicit', False),
+    ])
+    def test_broadcast(self, data, target_shape, ref_out, axes_mapping=None, mode='numpy', val_broadcast=True):
```

Comment:
I would like to see a couple but hand-made tests for broadcasting. The existing  is too simple because the input data is just a tensor of shape [1]. We should provide input data of shape [1, 2], [3, 1], [2, 1, 3] or similar with **different values** inside to make sure that broadcasting work as expected.

Reply:
Agree, added nontrivial unit-test cases.